packages/purpose-process/src/services/purposeService.ts

```diff
@@ -152,6 +159,50 @@ export function purposeServiceBuilder(
 
       return retrievePurposeVersionDocument(purposeId, version, documentId);
     },
+    async deletePurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<void> {
+      logger.info(`Deleting Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      if (organizationId !== purpose.data.consumerId) {
+        throw organizationIsNotTheConsumer(organizationId);
+      }
+
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      if (
+        purposeVersion.state !== purposeVersionState.waitingForApproval ||
+        purpose.data.versions.length === 1
```

Comment:
this could be a `isDeletable`

Reply:
Updated (I used `isDeletableVersion` to avoid confusion with `deletePurpose`)